src/NextjsRevalidation.ts

```diff
@@ -27,25 +31,34 @@ export interface NextjsRevalidationProps extends NextjsBaseProps {
 }
 
 /**
- * Builds the system for revalidating Next.js resources. This includes a Lambda function handler and queue system.
+ * Builds the system for revalidating Next.js resources. This includes a Lambda function handler and queue system as well
+ * as the DynamoDB table and provider function.
  *
  * @see {@link https://github.com/serverless-stack/open-next/blob/main/README.md?plain=1#L65}
  *
  */
 export class NextjsRevalidation extends Construct {
   queue: Queue;
-  function: LambdaFunction;
+  table: Table;
+  queueFunction: LambdaFunction;
+  tableFunction: LambdaFunction | undefined;
   private props: NextjsRevalidationProps;
 
   constructor(scope: Construct, id: string, props: NextjsRevalidationProps) {
     super(scope, id);
     this.props = props;
 
     this.queue = this.createQueue();
-    this.function = this.createFunction();
+    this.queueFunction = this.createFunction();
 
-    // allow server fn to send messages to queue
-    props.serverFunction.lambdaFunction?.addEnvironment('REVALIDATION_QUEUE_URL', this.queue.queueUrl);
+    this.table = this.createRevalidationTable();
+    this.tableFunction = this.createRevalidationInsertFunction(this.table);
+
+    this.props.serverFunction.lambdaFunction.addEnvironment('CACHE_DYNAMO_TABLE', this.table.tableName);
+    this.table.grantReadWriteData(this.props.serverFunction.lambdaFunction.role!);
```

Comment:
If for whatever reason the role is `undefined`, I don't want this to error. Could you change to:
```ts
if (this.props.serverFunction.lambdaFunction.role) {
  this.table.grantReadWriteData(this.props.serverFunction.lambdaFunction.role);
}
```